### ISSUE TITLE
netMHCstabpan not filtering invalid alleles before submission.

### ISSUE THREAD

**Issue by eddieimada** [griffithlab/pVACtools#848]
version: latest full docker version

If an HLA allele that is supported by pvacseq but not by netMHCstabpan, pvacseq fails to filter invalid alleles causing an error.

**To Reproduce**

singularity run resources/pvactools.sif pvacseq netmhc_stab input_file  output_file
*the error occurs whether you execute netMHCstabpan during pvacseq or afterwards.

**Log Output**
```
Traceback (most recent call last):
  File "/usr/local/bin/pvacseq", line 8, in <module>
    sys.exit(main())
  File "/usr/local/lib/python3.7/site-packages/pvactools/tools/pvacseq/main.py", line 116, in main
    args[0].func.main(args[1])
  File "/usr/local/lib/python3.7/site-packages/pvactools/tools/pvacseq/netmhc_stab.py", line 12, in main
    NetMHCStab(args.input_file, args.output_file, file_type='pVACseq', top_score_metric=args.top_score_metric).execute()
  File "/usr/local/lib/python3.7/site-packages/pvactools/lib/netmhc_stab.py", line 127, in execute
    raise Exception("Unexpected return value from NetMHCstabpan server. Unable to parse response.\n{}".format(response.content.decode()))
Exception: Unexpected return value from NetMHCstabpan server. Unable to parse response.
<BODY LINK="#0000ff" BGCOLOR="#ffffff">

<script language=JavaScript type = "text/javascript">
  var loc = window.top.location.search.substring(1);
  if (loc != null && loc !== "" && loc.length > 1) {
      document.write('<center><h2>',loc,' Server Output - DTU Health Tech</h2></center>');}
  </script>
<hr>
<pre>

# NetMHCstabpan version 1.0

# Input is in FSA format

# Peptide length 8
Error. HLA-B39:90 cannot be found in hla_pseudo list /tools/src/netMHCstabpan-1.0/Linux_x86_64/data/MHC_pseudo.dat
</pre>
<hr>
Go <a href="javascript:history.back()"><b>back</b></a>.
<!--
<button type="button"
onclick="window.top.location.reload(true);")>
New Submission</button>
-->
</BODY>
</HTML>
```

**Expected behavior**
Ignore the invalid alleles and add the information for the valid ones.

Input file:
[Scrn11T.txt](https://github.com/griffithlab/pVACtools/files/9746137/Scrn11T.txt)

**Comment by susannasiebert** [comment]
Thank you for informing us about this bug. NetMHCstabpan usually attempts to find the closest neighbor to the selected HLA allele if the allele is not directly supported. It appears that NetMHCstabpan does not have a lookup entry for this particular allele. We should handle this error more gracefully by not annotating these entries instead of erroring out. I will add this issue to our next issue milestone.

You might want to consider contacting the team behind the NetMHCstabpan server (https://services.healthtech.dtu.dk/service.php?NetMHCstabpan-1.0) to see if your allele can be added.

Note to self: a list of supported HLA alleles can be found [here](https://services.healthtech.dtu.dk/service.php?NetMHCstabpan-1.0)

**Comment by eddieimada** [comment]
Dear Susanna,

Thank you for the quick reply. 

Would it be reasonable to add aliases to the closest neighbor to fallback when running netMHCstabpan directly in pvacseq? Maybe a flag to turn an option like this? 

I will contact the netMHCstabpan team, but I believe there will always be discrepancies between support alleles between the tools. So maybe this might be a solution.

**Comment by eddieimada** [comment]
Apologies, closed by mistake,

**Comment by susannasiebert** [comment]
That's an interesting idea but I don't know on the top of my head how we would go about determining the closest neighbor. I would need to do some research to see if there are tools available that handle this for all alleles. NetMHCstabpan does this work for most alleles and will output results for them so we haven't had to worry about this before.

**Comment by susannasiebert** [comment]
This issue should be fixed in version 3.1.0. Please give it a try and let me know if you're running into any additional problems.
